src/ClientTransactionTask.h

```diff
@@ -0,0 +1,264 @@
+/* Copyright (c) 2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef RAMCLOUD_CLIENTTRANSACTIONTASK_H
+#define RAMCLOUD_CLIENTTRANSACTIONTASK_H
+
+#include <list>
+#include <map>
+#include <memory>
+
+#include "RamCloud.h"
+
+namespace RAMCloud {
+
+/**
+ * The Client Transaction Task represents an ongoing or preparing transaction
+ * commit.  This module is responsible for driving a client requested
+ * transaction to completion.  It also allows the commit process to be handled
+ * asynchronously by the RamCloud client library on the clients behalf.  For
+ * efficiency, this module also stores the pre-commit values (the commit cache)
+ * for the client transaction module.
```

Comment:
The comment above is pretty scattered; it just reads like a bunch of random ideas that is hard to see how they fit together. How about this is an alternative:

This class manages the state of a Transaction. It caches information about all of the operations that are part of the transaction (reads, writes, deletes), and it executes the client-driven protocol for committing the transaction. Furthermore, itallows the commit protocol to be executed asynchronously.